internal/handlers/images.go

```diff
@@ -98,12 +99,23 @@ func NewImageHandler(is imagestore.ImageStore, assistedServiceScheme, assistedSe
 		GenerateImageStream:   isoeditor.NewRHCOSStreamReader,
 		RequestAuthType:       requestAuthType,
 		Client:                client,
+		sem:                   make(chan struct{}, maxRequests),
 	}
 
 	return stdmiddleware.Handler("/images/:imageID", mdw, h)
 }
 
 func (h *ImageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	select {
```

Comment:
Thoughts on extracting this to a function or middleware of some sort?

I didn't on this first pass because I don't think it's going to realistically be reused anywhere.

Reply:
I think it's fine where it is.